src/main/java/net/openhft/chronicle/testframework/internal/process/InternalJavaProcessBuilder.java

```diff
@@ -16,37 +18,55 @@
 import java.util.List;
 import java.util.stream.Collectors;
 
-public final class InternalProcessRunner {
+public final class InternalJavaProcessBuilder implements JavaProcessBuilder {
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(InternalProcessRunner.class);
+    private static final Logger LOGGER = LoggerFactory.getLogger(InternalJavaProcessBuilder.class);
 
-    private InternalProcessRunner() {
-        // Suppresses default constructor, ensuring non-instantiability.
+    private final Class<?> mainClass;
+    private String[] jvmArguments;
+    private String[] programArguments;
+    private String[] classpathEntries;
+    private boolean inheritIO = false;
+
+    public InternalJavaProcessBuilder(@NotNull final Class<?> mainClass) {
+        this.mainClass = mainClass;
     }
 
-    /**
-     * Spawn a process running the main method of a specified class
-     *
-     * @param clazz The class to execute
-     * @param args  Any arguments to pass to the process
-     * @return the Process spawned
-     * @throws IOException if there is an error starting the process
-     */
-    public static Process runClass(Class<?> clazz, String... args) throws IOException {
-        return runClass(clazz, new String[]{}, args, null);
+    @Override
+    public InternalJavaProcessBuilder withProgramArguments(String... programArguments) {
+        this.programArguments = programArguments;
+        return this;
+    }
+
+    @Override
+    public InternalJavaProcessBuilder withJvmArguments(String... jvmArguments) {
+        this.jvmArguments = jvmArguments;
+        return this;
+    }
+
+    @Override
+    public InternalJavaProcessBuilder withClasspathEntries(String... classpathEntries) {
+        this.classpathEntries = classpathEntries;
+        return this;
+    }
+
+    @Override
+    public InternalJavaProcessBuilder inheritingIO() {
+        this.inheritIO = true;
+        return this;
+    }
+
+    @Override
+    public Runner build() {
+        return this::start;
```

Comment:
Is there much value in separating the build and start actions? I could imagine it might be useful if we wanted to create a few processes from a template, that differed slightly in their parameters, but that wouldn't even be possible the way things are because the returned Runner will reference the fields from the builder. I'd be pro removing this and just having start on the builder unless there's some utility the split provides.
I guess the only advantage at the moment is you could pass the Runner to another piece of code and know it wasn't going to mess with the arguments you configured?

Reply:
It is more that a Builder should always have a `build()` method and so, this is less surprising than allowing the Builder to be "active". I will take a look at fixing the other problems you mention.

Reply:
I had really considered it a specialised form of ProcessBuilder which does have a start method. It seems from the javadoc that ProcessBuilder is built with potentially many calls to start in mind.

https://docs.oracle.com/javase/7/docs/api/java/lang/ProcessBuilder.html

Reply:
I am open to changing this as per your suggestion @nicktindall

Reply:
Thanks @minborg I have made the change, if we see a good use case for separating `build()` and `run()` we can add it in the future and just have `start()` as shorthand for `build().run()`